keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/ResourceHandler.kt

```diff
@@ -75,6 +75,17 @@ interface ResourceHandler<S : ResourceSpec, R : Any> : SpinnakerExtensionPoint {
    */
   suspend fun current(resource: Resource<S>): R?
 
+  /**
+   * Evaluates the resource diff and determines whether it can take action to resolve the diff.
+   * @return a response indicating if it can take action, and a message explaining why if it can't.
+   *
+   * The default value is to take action on all changes. Only override if needed.
+   */
+  suspend fun willTakeAction(
```

Comment:
it's a bit confusing to see `willTakeAction` as both a function and a property in `ActionResponse`. Is it possible to rename one of those?

Reply:
Ha yes very true. Do you have any ideas? I am very bad at naming

Reply:
me too 😅  maybe `isActionRequired?`